core/src/main/scala/chisel3/ModuleImpl.scala

```diff
@@ -671,8 +688,9 @@ package experimental {
         // PseudoModules are not "true modules" and thus should share
         // their original modules names without uniquification
         this match {
-          case _: PseudoModule => desiredName
-          case _ => Builder.globalNamespace.name(desiredName)
+          case _: PseudoModule => Module.currentModulePrefix + desiredName
+          case _: BaseBlackBox => Builder.globalNamespace.name(desiredName)
+          case _ => Builder.globalNamespace.name(Module.currentModulePrefix + desiredName)
```

Comment:
```suggestion
          case _: PseudoModule => desiredName
          case _: Prefixable => Builder.globalNamespace.name(Module.currentModulePrefix + desiredName)
          case _ => Builder.globalNamespace.name(desiredName)
```
I don't care what we call it, but using `Prefixable` as a strawman, have `RawModule` mix the trait in, then users can decide to opt-in their BlackBoxes or go with the default behavior.

Also note I removed the prefix from `Pseudo-module`, I'm pretty sure they shouldn't be getting the prefix (but maybe I'm wrong).

Reply:
Discussed offline that we are leaving this proposal to a future PR.